lldb/source/Plugins/ObjectFile/Mach-O/ObjectFileMachO.cpp

```diff
@@ -5669,6 +5673,10 @@ bool ObjectFileMachO::GetCorefileMainBinaryInfo(addr_t &value,
             type = eBinaryTypeStandalone;
             typestr = "standalone";
             break;
+          case 4:
+            type = eBinaryTypeUserAllImageInfos;
+            typestr = "userland dyld_all_image_infos";
+            break;
```

Comment:
Should this have a default case that sets `type` to `eBinaryTypeInvalid`?

Reply:
If you do, it might be useful to add a log. If that situation were to happen, you'd see it in the logs immediately and know that either the core file looks unexpected or LLDB has a bug in core file handling.

Reply:
The `typestr` is set to `"unrecognized type"` and will be printed if logging is enabled, but the `type` enum is uninitialized and will also be printed, which could be confusing.  I'll add an initialization.  This method doesn't currently take a Target or Process pointer so I can't print a message to the user asynchronously, it'll require them to enable logging when something is going wrong to find it.